src/SqlStreamStore.Postgres/PostgresStreamStore.ReadAll.cs

```diff
@@ -195,18 +244,15 @@ private async Task PollTransactions(Guid correlation, TxIdList transactionsInPro
             CancellationToken cancellationToken)
         {
             Logger.TraceFormat("Correlation: {correlation} | Read trusted message initiated", correlation);
-
             var (messages, maxAgeDict, _, isEnd) = await ReadAllForwards(fromPositionInclusive, maxCount, prefetch, correlation, cancellationToken).ConfigureAwait(false);
-            
-            Logger.TraceFormat("Correlation: {correlation} | Filter messages from {fromPositionInclusive} to {toPositionInclusive}", correlation, fromPositionInclusive, toPositionInclusive);
 
+            Logger.TraceFormat("Correlation: {correlation} | Filter messages from {fromPositionInclusive} to {toPositionInclusive}", correlation, fromPositionInclusive, toPositionInclusive);
             var messageToReturn = messages.Where(x => x.Position >= fromPositionInclusive && x.Position <= toPositionInclusive).ToList();
 
-            if(isEnd && messageToReturn.Count <= messages.Count)
+            if(isEnd && messageToReturn.Count != messages.Count)
```

Comment:
@ArneSchoonvliet Can you really think about this change ;)

Imho 'messageToReturn.Count' is always lower or equal to messages.Count
So it would always get into the code.

It could also be changed to
`messages.Count > messageToReturn.Count`

(But since we filter out values != is actually the same as 'greater than' if you understand what I mean)